pkg/activator/handler/healthz_handler_test.go

```diff
@@ -77,3 +78,49 @@ func TestHealthHandler(t *testing.T) {
 		})
 	}
 }
+
+func BenchmarkHealthHandler(b *testing.B) {
+	examples := []struct {
+		name    string
+		headers http.Header
+		check   func() error
+	}{{
+		name:    "forward non-kubelet request",
+		headers: mapToHeader(map[string]string{"User-Agent": "chromium/734.6.5"}),
+		check:   func() error { return nil },
+	}, {
+		name:    "kubelet probe success",
+		headers: mapToHeader(map[string]string{"User-Agent": "kube-probe/something"}),
+		check:   func() error { return nil },
+	}, {
+		name:    "kubelet probe failure",
+		headers: mapToHeader(map[string]string{"User-Agent": "kube-probe/something"}),
+		check:   func() error { return errors.New("not ready") },
+	}}
+
+	logger := ktesting.TestLogger(b)
+	baseHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
+	req := httptest.NewRequest(http.MethodGet, "http://example.com", nil)
+	for i, e := range examples {
+		handler := HealthHandler{HealthCheck: e.check, NextHandler: baseHandler, Logger: logger}
+		req.Header = e.headers
+		resp := httptest.NewRecorder()
```

Comment:
if you move the `resp =` call below, let's move this into the sequential bracket for consistency.